invenio/modules/deposit/models.py

```diff
@@ -521,8 +553,29 @@ def __unicode__(self):
         """ Return a name for this class """
         return self.get_identifier()
 
+    @classmethod
+    def all_authorized(cls, current_user):
+        """
+        Return a dict of deposit types that the current user
+        is allowed to view.
+        """
+        from invenio.modules.access.control import acc_get_action_id, \
+            acc_find_possible_actions_user
+
+        all_types = cls.all()
+        action_id = acc_get_action_id('usedeposit')
+
+        avail_actions = acc_find_possible_actions_user(current_user.get_id(),
+                                                       action_id)[1:]
+        values = [s for s in itertools.chain(*avail_actions)
+                  if isinstance(s, basestring)]
+
+        # filter the all_types dict()
+        return {key: all_types[key] for key in values if key in all_types}
```

Comment:
@jirikuncar I changed the user_info line that you suggested. Regarding:

`return {key: all_types[key] for key in all_types.keys() if acc_authorize_action(usedeposit, type=key)}`

I think this is semantically the same as code above, but little cleaner and more readable. However it would require much more database queries, which might hurt the performance. If you think it's better I may change it.

Can you also explain me how do you calculate the `precached_allowed_deposition_types`? I didn't manage to find anything similar in the code. Is it a user property? When should it be calculated?